cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_csr_impl_dpc.cpp

```diff
@@ -0,0 +1,180 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel.hpp"
+#include "oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_csr_impl.hpp"
+#include "oneapi/dal/table/csr_accessor.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/util/common.hpp"
+#include "oneapi/dal/detail/policy.hpp"
+#include "oneapi/dal/backend/communicator.hpp"
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+namespace oneapi::dal::basic_statistics::backend {
+
+namespace de = dal::detail;
+namespace bk = dal::backend;
+namespace pr = dal::backend::primitives;
+
+using method_t = method::sparse;
+using task_t = task::compute;
+using comm_t = bk::communicator<spmd::device_memory_access::usm>;
+using input_t = compute_input<task_t, dal::csr_table>;
+using result_t = compute_result<task_t>;
+using descriptor_t = detail::descriptor_base<task_t>;
+
+template <typename Float>
+result_t compute_kernel_csr_impl<Float>::operator()(const bk::context_gpu& ctx,
+                                                    const descriptor_t& desc,
+                                                    const input_t& input) {
+    auto queue = ctx.get_queue();
+    const csr_table csr_tdata = input.get_data();
+    const auto column_count = csr_tdata.get_column_count();
+    const auto row_count = csr_tdata.get_row_count();
+    auto result_options = desc.get_result_options();
+    const auto nonzero_count = csr_tdata.get_non_zero_count();
+    auto [csr_data, column_indices, row_offsets] =
+        csr_accessor<const Float>(csr_tdata).pull(queue, { 0, -1 }, sparse_indexing::zero_based);
+    auto csr_data_ptr = csr_data.get_data();
+    auto column_indices_ptr = column_indices.get_data();
+
+    using limits_t = std::numeric_limits<Float>;
+    constexpr Float maximum = limits_t::max();
+
+    auto result_data = pr::ndarray<Float, 2>::empty(queue,
+                                                    { res_opt_count_, column_count },
+                                                    sycl::usm::alloc::device);
+    auto result_data_ptr = result_data.get_mutable_data();
+
+    const auto local_size = std::min(bk::device_max_wg_size(queue), bk::down_pow2(nonzero_count));
+    const auto nd_range =
+        bk::make_multiple_nd_range_2d({ column_count, local_size }, { 1, local_size });
+    auto event = queue.submit([&](sycl::handler& cgh) {
```

Comment:
Agree with @KulikovNikita that more efficient version may be implemented. Unnecessary indices checks can be removed.
Here is an idea in C++:
```cpp
    std::array<float, column_count> min;
    std::array<float, column_count> sum;
    min.fill(std::numeric_limits<float>::max());
    sum.fill(0.0f);

    for (int icsr = row_offsets[0]; icsr < row_offsets[row_count]; ++icsr) {
        const int icol = column_indices[icsr];
        const float val = data[icsr];

        min[icol] = std::min<float>(min[icol], val);
        sum[icol] += val;
    }
```

Reply:
This is for zero-based indices.

Reply:
4. and 5. are not thread safe by this meaning. It requires at least atomic synchronization among all working items to access same column index.
So, I have 2 ways to implement it: 
1. Follow your suggestion and add atomics to sync with global memory
2. Make working item responsible for exact column (no need to sync, because n_working_items = column_count).
Now the second is implemented.
I will definitely try both cases when benchmarks are  ready.
If you 100% sure, that atomic blocking will work faster, I will do it in the scope of this PR. 
Please, respond